### PR TITLE
Optimize function getIPThroughDHCP

### PR DESCRIPTION
**Pull request by yaocw2020**
issue: https://github.com/harvester/harvester/issues/1209

Use the nclient4 package to get the IP through DHCP instead of executing the wicked command.

### CODE REVIEW

**Review by guangbochen — APPROVED** [review]
LGTM

**Comment by gitlawr** [comment]
Can you explain why this solution resolves the issue?

**Comment by yaocw2020** [comment]
> Can you explain why this solution resolves the issue?

There are two reasons:
- To avoid parsing wicked output.
- Generally, the DHCP server should have a record about the vip after installation before kube-vip applying, but `wickedd-dhcp4` does not. It may cause kube-vip fail to requests the vip.

File: pkg/console/vip.go [hunk 1]
```diff
@@ -21,21 +25,23 @@ func createMacvlan(name string) (netlink.Link, error) {
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to fetch %s", name)
 	}
+	rand.Seed(time.Now().UnixNano())
+	macvlanName := tempMacvlanPrefix+strconv.Itoa(rand.Intn(100))
```
Comment (gitlawr) [inline pkg/console/vip.go:29]: What happens if we have a conflict name?  A random integer in 100 doesn't look safe

Comment (yaocw2020) [reply pkg/console/vip.go:29]: I think 100 is far enough. After requesting the VIP, the temporary macvlan device will be deleted immediately. Unless users add network interfaces on purpose, there would not have a conflict network interface. With a very small probability, there will be an error message on the console if a conflict occurs.

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by bk201 — APPROVED** [review]
The change also fixes the problem that I can't get VIP from dnsmasq after rebooting.

**Pull request merged**